### PR TITLE
Fix(Modal): adds tabIndex so user can scroll with arrow key

### LINKED ISSUE #3261: AVT 2 (webkit) - React Transactional Modal & Input Modal -  Keyboard users can't scroll content
## Environment
macOS Mojave version 10.14.5
Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React

## Detailed Description
Tab to the Launch Modal button 
Arrow keys do not scroll the content

### PR DESCRIPTION
**Pull request by abbeyhrt**
Closes #3261 

This PR adds `tabIndex="0"` to the modal content so that the text area can receive focus and be scrollable with arrow keys when necessary.  This seems to be an issue with webkit browsers that was supposedly fixed [here](https://groups.google.com/a/chromium.org/forum/#!msg/blink-dev/Fku6784p0wI/3MyOMTk7BwAJ) but we aren't seeing that. Let me know if the aria-labelledby needs to be changed, the label already supplied to the modal seemed sufficient but I'm not sure! 

used this: https://developer.paciellogroup.com/blog/2016/02/short-note-on-improving-usability-of-scrollable-regions/ as guidance 
#### Changelog

**Changed**

- tabIndex added to modal body content

#### Testing / Reviewing

Go to the ModalWrapper -> transactional/passive modal open the modal and tab to the content, make sure it receives focus and that you can use the arrow keys to scroll

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
Thank you for tacking another issue in this problem space @abbeyhrt! One question - How does the focus-wrap behavior works with this change?

**Comment by dakahn** [comment]
@asudoh looks like it just adds another tab stop to the focus wrap.

**Review by dakahn — APPROVED** [review]

File: packages/react/src/components/Modal/Modal.js [hunk 1]
```diff
@@ -345,7 +345,14 @@ export default class Modal extends Component {
           <h3 className={`${prefix}--modal-header__heading`}>{modalHeading}</h3>
           {!passiveModal && modalButton}
         </div>
-        <div className={`${prefix}--modal-content`}>{this.props.children}</div>
+        <div
```
Comment (joshblack) [inline packages/react/src/components/Modal/Modal.js:348]: Should we be adding this dynamically when we know the content has overflowed?

**Review by joshblack — CHANGES_REQUESTED** [review]

**Comment by asudoh** [comment]
@dakahn Yeah that's what I imagined. In case I wasn't clear enough; My question is more about if it'd be desirable for users.

**Comment by dakahn** [comment]
@asudoh Without this functionality we're in violation of [principle 2.1.1](https://www.ibm.com/able/guidelines/ci162/keyboard.html) -- since the _as is_ is a scrollbar operable only via mouse. The scrollable content order makes sense. 

Unless you're talking about what @joshblack brought up -- where this keyboard focusability _always on_ and should probably only be there if we have scrollable content.

**Comment by asudoh** [comment]
I can see your point @dakahn. Another question; Does this problem reproduce with a focusable element in the modal content?

**Comment by emyarod** [comment]
I had open work for this to address #3816 but it looks like the ticket was closed as a dupe yesterday. there might be some overlap with what I have already but I opened up https://github.com/carbon-design-system/carbon/pull/3943 to check in what I currently have

**Comment by elizabethsjudd** [comment]
I brought up the point of dynamically checking for the content scrolling as well when I did the vanilla version and was told not to add that.  If that decision is changed can you make an issue to update vanilla as well and assign it to me (we have code in WH that we can port over).

**Comment by abbeyhrt** [comment]
@emyarod PR has a larger scope and addresses this problem in multiple places so I'm happy to go with his PR instead of this one, we can decide here or there if we want the tabIndex to be applied dynamically.

**Comment by dakahn** [comment]
@elizabethsjudd Can you talk about what reason you were given to _not_ check for scrollable content?

**Comment by elizabethsjudd** [comment]
@dakahn https://github.com/carbon-design-system/carbon/issues/3196#issuecomment-517453125

**Comment by joshblack** [comment]
> we can decide here or there if we want the tabIndex to be applied dynamically.

It seems like we'd want to offer this in terms of the experience, otherwise we'd rely on teams having to implement this logic each time they'd want to support this use-case. Seems like something we could offer by default (although haven't explored what the full implementation should look like)

**Comment by joshblack** [comment]
Side-note, did we even want to implement this? It seemed like a webkit specific bug, wasn't it @dakahn? At least going through the reported webkit bugs and the implementors seemed to agree FF was implementing to spec.

**Comment by abbeyhrt** [comment]
Closing since it's covered in #3943

**Pull request closed without merging**